src/parse-date.js

```diff
@@ -3,6 +3,54 @@
 var expirationYear = require('./expiration-year');
 var isArray = require('./lib/is-array');
 
+function monthLength(value) {
```

Comment:
let's change this function name to be more explicit about what it does. Maybe `getNumberOfMonthDigitsInDateString`. It's perhaps too verbose, but it makes it absolutely clear what it is doing.

And let's also change `value` to `dateString`.

```js
function getNumberOfMonthDigitsInDateString(dateString) {
```

Reply:
Also, big fan of pulling this logic out into a function and explaining what each bit means. :+1:

Reply:
I treat this only positively 😋